File: 053_Maximum_Subarray.cpp
int maxSubArray(int* nums, int numsSize){
    int sum = 0;
    int maxsum = -10000;
    for(int i=0;i<numsSize;i++){
        if(sum>0){
            sum += nums[i];
        }
        else{
            sum = nums[i];
        }
        if(sum > maxsum){
            maxsum = sum;
        }
    }
    return maxsum;
}

File: 053_Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum_ = 0
        maxsum = -10000
        for i in range(len(nums)):
            if sum_ > 0:
                sum_ += nums[i]
            else:
                sum_ = nums[i]
            if sum_ > maxsum:
                maxsum = sum_
        return maxsum
